### PR TITLE
Fix case where one detector combo has no foreground triggers

### PR DESCRIPTION
**Pull request by connor-mcisaac**
The add statmap code is breaking when finding the most significant trigger for hierarchical removal if there are no triggers in a given detector combination.

In the dictionary comprehension it does argmax for each detector combo, if one is empty this produces an error.
This can be fixed by including an if statement in the comprehension to test that the array is non-zero in size.

### CODE REVIEW

**Review by GarethDaviesGW — COMMENTED** [review]
This looks good and fixes the issue - as its doing the same loop twice it _might_ be preferable to have a normal loop which adds to a dictionary, but dont think it really matters

Have you tested that nothing later in the code relies on the keys of either dictionary for the full list of combinations?

File: bin/hdfcoinc/pycbc_multiifo_add_statmap [hunk 1]
```diff
@@ -453,9 +453,11 @@ while True:
                     combined_fg_data.data[key][:]
 
     max_each_combo = {combo: sep_fg_data[combo].data['ifar'][:].argmax()
-                      for combo in all_ifo_combos}
+                      for combo in all_ifo_combos
+                      if len(sep_fg_data[combo].data['ifar'][:]) > 0}
     max_ifars = {combo: sep_fg_data[combo].data['ifar'][:][max_each_combo[combo]]
-                    for combo in all_ifo_combos}
+                 for combo in all_ifo_combos
```
Comment (GarethDaviesGW) [inline bin/hdfcoinc/pycbc_multiifo_add_statmap:459]: If this is updated to `for combo in max_each_combo.keys()` that would make it a bit more readable

**Comment by connor-mcisaac** [comment]
> This looks good and fixes the issue - as its doing the same loop twice it _might_ be preferable to have a normal loop which adds to a dictionary, but dont think it really matters
> 
> Have you tested that nothing later in the code relies on the keys of either dictionary for the full list of combinations?

After this point these 2 dicts are used to get the loudest trigger and then are not used after that.

File: bin/hdfcoinc/pycbc_multiifo_add_statmap (hunk 1 above)
Comment (connor-mcisaac) [reply bin/hdfcoinc/pycbc_multiifo_add_statmap]: I've updated to use the keys and value of the previous dict to make it simpler

**Review by GarethDaviesGW — APPROVED** [review]
Looks good, thanks!

**Pull request merged**